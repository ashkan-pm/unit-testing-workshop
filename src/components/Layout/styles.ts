import styled from 'styled-components';

export const LayoutWrapper = styled.main`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  min-height: 100vh;
`;
