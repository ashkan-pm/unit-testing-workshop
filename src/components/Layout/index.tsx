import React from 'react';
import { LayoutWrapper } from './styles';

type Props = {
  children: React.ReactNode;
};
function Layout({ children }: Props) {
  return <LayoutWrapper>{children}</LayoutWrapper>;
}

export default Layout;
