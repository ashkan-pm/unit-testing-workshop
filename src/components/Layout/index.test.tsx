import React from 'react';
import { render, screen } from '@testing-library/react';
import Layout from './index';

test('Layout renders correctly', () => {
  render(<Layout />);
  const layout = screen.getByRole('main');
  expect(layout).toMatchSnapshot();
});
