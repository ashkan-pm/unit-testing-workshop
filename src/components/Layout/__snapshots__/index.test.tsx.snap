// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

<main
  class="c0"
/>
`;
